File: src/main/kotlin/aurora/jwt/common/util/ValidationUtils.java
package aurora.jwt.common.util;

public final class ValidationUtils {
    private ValidationUtils() {}

    public static void checkArgument(boolean expression, String errorMessage) {
        if (!expression) {
            throw new IllegalArgumentException(errorMessage);
        }
    }

    public static String notNullOrEmpty(String str, String errorMessage) {
        checkArgument(str != null && !str.isEmpty(), errorMessage);
        return str;
    }
}

File: src/main/kotlin/aurora/jwt/common/util/ValidationUtils.kt
package aurora.jwt.common.util

object ValidationUtils {
    fun checkArgument(expression: Boolean, errorMessage: String) {
        require(expression) { errorMessage }
    }

    fun notNullOrEmpty(str: String?, errorMessage: String): String {
        checkArgument(str != null && !str.isEmpty(), errorMessage)
        return str!!
    }
}
